scripts/update_subgraph_counts.py

```diff
@@ -0,0 +1,92 @@
+"""
+Count the number of subgraphs per configured source and produce output to
+expedite updated source configurations.
+"""
+
+import argparse
+import sys
+
+import attr
+
+from azul import (
+    config,
+)
+from azul.args import (
+    AzulArgumentHelpFormatter,
+)
+from azul.azulclient import (
+    AzulClient,
+)
+from azul.indexer import (
+    Prefix,
+)
+from azul.openapi import (
+    format_description,
+)
+from azul.terra import (
+    TDRSourceSpec,
+)
+
+parser = argparse.ArgumentParser(description=__doc__,
+                                 formatter_class=AzulArgumentHelpFormatter)
+
+parser.add_argument('--catalogs',
+                    nargs='+',
+                    metavar='NAME',
+                    default=[
+                        c for c in config.catalogs
+                        if c not in config.integration_test_catalogs
+                    ],
+                    help='The names of the catalogs to determine source specs for.')
+
+
+@attr.s(auto_attribs=True, frozen=True, kw_only=True)
+class SourceSpecArgs:
+    project: str
+    snapshot: str
+    subgraph_count: int
+
+    def __str__(self) -> str:
+        return f'mksrc({self.project!r}, {self.snapshot!r}, {self.subgraph_count!r})'
+
+
+azul = AzulClient()
```

Comment:
We prefer not to have these kinds of side effects in the top-level of scripts. This can be a local of the `generate_sources` method.